test/functional/tests/lazy_writes/wb_throttling.py

```diff
@@ -0,0 +1,153 @@
+from storage_devices.disk import DiskType, DiskTypeSet, DiskTypeLowerThan
+from test_tools import fs_utils
+from test_tools.fio.fio import Fio
+from test_tools.fio.fio_param import ReadWrite, IoEngine
+from test_tools.iostat import IOstat
+from test_utils.os_utils import (
+    kill_all_io,
+    set_wbt_lat,
+    get_wbt_lat,
+    get_dut_cpu_number,
+)
+from api.cas import casadm
+from api.cas.cache_config import (
+    CacheMode,
+    CacheModeTrait,
+    CleaningPolicy,
+    FlushParametersAcp,
+    SeqCutOffPolicy,
+    CacheLineSize,
+    Time,
+)
+from test_tools.blktrace import BlkTrace, BlkTraceMask, ActionKind, RwbsKind
+
+
+runtime = 60 * 60 * 24 * 30
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_wb_throttling():
+    """
+        title: Test CAS with write back throttling enabled on exported object
+        description: |
+          Fill cache with data, run intensive IO (rwmix=70) with occasionals trims and wait for
+          hang task or kernel panic.
+        pass_criteria:
+          - Hang task did not occurred
+          - System did not crashed
+    """
+    with TestRun.step("Prepare devices."):
+        cache_device = TestRun.disks["cache"]
+        core_device = TestRun.disks["core"]
+        set_wbt_lat(cache_device, 0)
+        set_wbt_lat(core_device, 0)
+        if core_device.size > cache_device.size:
+            TestRun.LOGGER.info("Starting cache on partition")
+            cache_device.create_partitions([Size(core_device.size)])
+            cache_device = cache_device.partitions[0]
+
+    with TestRun.step(
+        f"Start cache with one core in write back with 64k cache line, NOP and disabled seq cutoff"
+    ):
+        cache = casadm.start_cache(cache_device, CacheMode.WB, CacheLineSize.LINE_64KiB)
+        cache.set_cleaning_policy(CleaningPolicy.nop)
+        core = cache.add_core(core_device)
+        cache.set_seq_cutoff_policy(SeqCutOffPolicy.never)
+
+    with TestRun.step("Check wbt_lat value on exported object"):
+        wbt_lat_val = get_wbt_lat(core)
+        TestRun.LOGGER.info(f"wbt_lat for exported object is {wbt_lat_val}")
+        if wbt_lat_val == 0:
+            TestRun.LOGGER.info(f"Setting wbt_lat for exported object to 75000us")
+            set_wbt_lat(core, 7500)
+
+    with TestRun.step("Fill cache with dirty data"):
+        fio = get_fio_rw_cmd(core, write_percentage=100)
+        fio_pid = fio.run_in_background()
+        i = 0
+        while core.get_statistics(percentage_val=True).usage_stats.dirty < 100:
+            i += 1
+            if i == 60:
+                TestRun.LOGGER.info("Waiting to fill cache with dirty data")
+                TestRun.LOGGER.debug(
+                    f"Dirtiness level "
+                    f"{cache.get_statistics(percentage_val = True).usage_stats.dirty}"
+                )
+                i = 0
+            time.sleep(1)
```

Comment:
you can use os_utils.wait(lambda: core.get_statistics(percentage_val=True).usage_stats.dirty == 100, ...) method here